### ISSUE TITLE
New driver support 537.70 Grid Windows

### ISSUE THREAD

**Issue by vincentdesbr** [keylase/nvidia-patch#746]
Hello, i use NVIDIA-GRID-Linux-KVM-535.129.03-537.70 drivers on a Proxmox host, i looking to patch the drivers for windows 11 guest. The drivers are 537.70_grid_win10_win11_server2019_server2022_dch_64bit_international.exe this is the version for Linux-KVM-535.129.03 host.
- Nvidia driver version: 537.70 for guest and for host 535.129.03
- GPU Model: GTX 1080 TI
- OS: Host Proxmox 8.1.3 Linux 6.5.11-7-pve. Host drivers (535.129.03) are patched to unlock vgpu following the polloloco vgpu-proxmox tutorial. With this card i currently have only one nvenc session available. I can DM a link to download drivers if required.

Thanks in advance.

**Comment by jailuthra** [comment]
@vincentdesbr what does your GPU show up as in the guest? Is it still GTX 1080 or some Tesla GPU?

I am asking because https://docs.nvidia.com/grid/16.0/grid-vgpu-release-notes-generic-linux-kvm/index.html does not list GTX 1080.

In general data-center GPUs already have unlocked NVENC sessions, so you don't need a patch for Tesla GPUs for example.

**Comment by vincentdesbr** [comment]
Yes GPU show a P40.

This case is specific because it's a gtx 1080 ti, spoof as a P40 for drivers. Vgpu works well, but nvenc is only one session.
Windows 11 guest :
![Screenshot_2023-12-28_10-49-50](https://github.com/keylase/nvidia-patch/assets/155060937/7cf706ed-ff93-4195-8a2e-d3f2816de4c2)
Proxmox Host
![vgpu01](https://github.com/keylase/nvidia-patch/assets/155060937/ef9f9513-7228-40f1-bf52-3a9a18caaca5)

**Comment by jailuthra** [comment]
@vincentdesbr I don't think the guest will require patching as it sees a Tesla GPU (so it should allow multiple sessions)

Have you applied the NVENC patch for the host (linux 535.129.03)? If no, please apply it - follow the steps in the README of this repo.

After applying, can you test if you are able to run > 5 nvenc sessions on the linux host? Steps are given in the [wiki](https://github.com/keylase/nvidia-patch/wiki/Verify-NVENC-patch#linux)

**Comment by vincentdesbr** [comment]
@jailuthra Thanks for your support. I just git clone the patch and follow the steps. When i try to patch i get :

`root@pve:~/patchnvenc/nvidia-patch# ./patch.sh 
Detected nvidia driver version: 535.129.03
libnvidia-encode.so
ERROR: cannot detect driver directory`

`root@pve:~/patchnvenc/nvidia-patch# ./patch.sh -c 535.129.03
SUPPORTED`

To give you details, i don't use the Nvidia drivers (https://international.download.nvidia.com/XFree86/Linux-x86_64/535.129.03/NVIDIA-Linux-x86_64-535.129.03.run) but (NVIDIA-Linux-x86_64-535.129.03-vgpu-kvm.run).

**Comment by jailuthra** [comment]
@vincentdesbr i am not sure about vgpu-kvm drivers. where did you download it from?

can you try to search for `libnvidia-encode.so` file in your host? it might be present in some different directory than what the `patch.sh` script currently expects.

**Comment by vincentdesbr** [comment]
@jailuthra 

Hello,

Thank you, i reply to your direct email, because i can't share in public Nvidia drivers.

The file used for the Proxmox 8.1.3 Host : *
The file used for the Windows 11 Guest : *

I try some commands to find the file :

root@pve:/lib/nvidia# find / -name libnvidia*
/usr/lib/x86_64-linux-gnu/libnvidia-ml.so.1
/usr/lib/x86_64-linux-gnu/libnvidia-vgxcfg.so
/usr/lib/x86_64-linux-gnu/libnvidia-vgpu.so
/usr/lib/x86_64-linux-gnu/libnvidia-vgpu.so.535.129.03
/usr/lib/x86_64-linux-gnu/libnvidia-ml.so
/usr/lib/x86_64-linux-gnu/libnvidia-vgxcfg.so.535.129.03
/usr/lib/x86_64-linux-gnu/libnvidia-ml.so.535.129.03
root@pve:/lib/nvidia#

**Comment by jailuthra** [comment]
@vincentdesbr from the driver you shared, looks like the host driver is missing support for NVENC, as I don't see the `libnvidia-encode.so` file after extracting the driver.

I don't think this I (or this patch) can help you if NVENC itself is missing in your setup.

**Comment by kesawi** [comment]
> @vincentdesbr from the driver you shared, looks like the host driver is missing support for NVENC, as I don't see the `libnvidia-encode.so` file after extracting the driver.
> 
> I don't think I (or this patch) can help you if NVENC itself is missing in your setup.

I don't believe that `libnvidia-encode.so` is installed on the host when vGPU drivers are used, only in the guest VM as I encounter the same issue. This is with applying the patch to the 535.129.03 gridd drivers in a Ubuntu 22.04.4 guest.
